### PR TITLE
F-PLN page visual rework

### PR DESCRIPTION
**Pull request by Lollo999**
## Summary of Changes
This PR is a follow up to the FPLN scroll fix, 
Made some changes to the F-PLN page to be more accurate to the real one:
1> FIXED : scroll skipping the "END OF F-PLN" line when scrolling 
2> FIXED : "SPD/ALT" indication is now fixed on top, making it easier to read the data
3> FIXED : Destination airport line is now fixed on the bottom of the page (if ERASE/INSRT line is not present)
4> ADDED : unit of measure [NM] on the last row waypoint distance from previous point.
5> FIXED : text size and color in the first and last point (DEP/DEST)
6> FIXED:  "From" text should only be present if the wp the aircraft is flying from is in the first row of the page
7> FIXED: now also the departure and arrival wp in the flight plan change color when the plan is being edited 
8> minor edits and code clean up

## Screenshots (if necessary)
BEFORE:
test_post_fix

AFTER:

Microsoft Flight Simulator 2020 12 04 - 10 40 49 01_Trim (2)

## References
online videos, Honeywell manual, and also compared by visual clues to the FSL one for some details. I did my best to make it as close as possible to the real one using the already present code base. Any feedback is very appreciated.

## Additional context
Tested in a couple of flights and using different PCs, no issues found so far related to this update

Discord username (if different from GitHub): Lorenzo Pinna#9154

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by MisterChocker** [comment]
Why does you before not show the magenta asterix? Thats already implemented

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_FlightPlanPage.js [hunk 1]
```diff
@@ -324,9 +353,9 @@ class CDUFlightPlanPage {
                                 CDUFlightPlanPage.ShowPage(mcdu, offset);
                             };
 
-                            for (let j = 0; i < rowsCount - 1; i++) {
+                            for (let j = 0; i < rowsCount; i++) {
                                 rows[2 * i + 1] = holdRows[j++];
-                                if (i == rowsCount - 2 || j >= holdRows.length) {
+                                if (i == rowsCount - 1 || j >= holdRows.length) {
```
Comment (MisterChocker) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_FlightPlanPage.js:358]:
```suggestion
                                if (i === rowsCount - 1 || j >= holdRows.length) {
```

**Review by MisterChocker — CHANGES_REQUESTED** [review]

**Comment by NathanInnes** [comment]
Add changelog if you got time

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_FlightPlanPage.js [hunk 2]
```diff
@@ -224,7 +247,13 @@ class CDUFlightPlanPage {
...
                         }
                         const distance = (waypoint === mcdu.flightPlanManager.getActiveWaypoint() ? waypoint.liveDistanceTo : waypoint.distanceInFP);
-                        rows[2 * i] = [airwayName, (index >= activeIndex || waypoint.ident === "(DECEL)" ? distance.toFixed(0) : "")];
+                        var dstnc;
+                        if (i === 4) {
+                            dstnc = distance.toFixed(0) + "NM";
+                        } else {
+                            dstnc = distance.toFixed(0);
+                        }
+                        rows[2 * i] = [(index === 0 && offset == 0) ? "FROM" : airwayName, ((index >= activeIndex || waypoint.ident === "(DECEL)") && i != 0 ? dstnc : i === 0 ? "SPD/ALT" : ""), i === 0 ? (isFlying ? "UTC" : "TIME") : ""];
```
Comment (davwheat) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_FlightPlanPage.js:256]:
```suggestion
                        rows[2 * i] = [(index === 0 && offset === 0) ? "FROM" : airwayName, ((index >= activeIndex || waypoint.ident === "(DECEL)") && i !== 0 ? dstnc : i === 0 ? "SPD/ALT" : ""), i === 0 ? (isFlying ? "UTC" : "TIME") : ""];
```

**Review by davwheat — CHANGES_REQUESTED** [review]

**Comment by Lollo999** [comment]
> Why does your before not show the magenta asterix? Thats already implemented. #2070
> The scroll thing is already implemented. #2069
> Or did you made changes to that?

My bad, the clip is from the #2069 i made before this, #2070 was merged soon after so i forgot to remake those videos, i think the most accurate way to see this working is in game, since is the only way to know exactly if the behaviour is correct

**Comment by NathanInnes** [comment]
Did you mean to close the PR?

**Pull request closed without merging**